### DISCUSSION TITLE
Null result causes TRPCClientError: Invalid response or stream interrupted

### DISCUSSION THREAD

**Discussion by cdunn07** [trpc/trpc#5919] [Q&A] [answered]
I'm using trpc 11 in a  Nextjs application and running into this issue. Anytime my backend returns a null value in a trpc route, I get the TRPCClientError mentioned in the title. The query then tries to refetch several times, even though the null value is indeed correct. I'm still new to trpc so I'm assuming I have something configured incorrectly. Has anyone else experienced this issue?

**Comment by curiousyuvi** [comment]
I encountered a similar issue in my TRPC project, we solved it by little hack of changing `return null;` with `return 0;` in the procedures.

This might not be a good solution, but it fixed the problem.

**Reply by Vandivier** [reply]
`export const NULL_RESULT_TRPC_INT = 0`
way more profesh now lol

**Comment by riccardolardi** [comment]
Encountering the same issue with TRPC 11 in a Next.js app

`TRPCClientError: Invalid response or stream interrupted at TRPCClientError.from (webpack-internal:///./node_modules/@trpc/client/dist/TRPCClientError.mjs:45:16) at eval (webpack-internal:///./node_modules/@trpc/client/dist/links/httpBatchStreamLink.mjs:139:102)`

**Comment by chakrihacker** [comment]
I am also facing this issue

**Comment by Eduardo-BarrosoS** [comment]
I am also with this same problem

**Comment by AskYous** [comment]
Other search results said to not return `null` in the procedure. I returned an object but I'm still facing this issue.

**Comment by titi-lima** [comment]
I've been facing this since porting from `httpBatchLink` to `httpBatchStreamLink` in v11; it's pretty annoying, as react-query keeps refetching on error and causing duplicated requests to the server. Should we file this as an issue instead?

**Comment by LorenzoCruccu** [comment]
same issue during production deployment on vercel, using trpc 11:

```
 Generating static pages (3/4) 
 << query  #1 marker.getAllMarkers  {
  input: undefined,
  result: i [TRPCClientError]: Invalid response or stream interrupted
      at i.from (/vercel/path0/.next/server/chunks/619.js:12:47740)
      at /vercel/path0/.next/server/chunks/619.js:12:54544
      at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
    meta: undefined,
    shape: undefined,
    data: undefined,
    [cause]: w [Error]: Invalid response or stream interrupted
```

**Reply by dieptx** [reply]
same issue here  on vercel

**Comment by KATT** [comment, marked as answer]
I think this issue was addressed with #5945.

If someone encounters this, please open a bug report. Make sure to include what link you're using to fetch data.

**Reply by 2-fly-4-ai** [reply]
This is still an issue.

**Reply by KATT** [reply]
@2-fly-4-ai / @prateekjain98 you're clearly experiencing issues, please create a bug report 🙏

**Reply by 2-fly-4-ai** [reply]
> I think this issue was addressed with #5945.
> 
> If someone encounters this, please open a bug report. Make sure to include what link you're using to fetch data.

Can we provide more detailed error reporting so we don't run into this vague issue?

It happens when you go beyond a size limit or when your function runs longer than the max serverless function time.

The issue is, with trpc, it will not allow me to upload more than 1mb, none of the tutorials, docs or anything work anymore. The experimental commands for next etc do nothing.

The old way of putting the config in your [trpc] API route is no longer accepted either. 

So basically you just have a bunch of people running around unable to solve the most basic of issues, like increasing a size limit. 

Or they are running into issues with max serverless function timeout limits reached. Google search pulls up nothing. All the LLM's provide incorrect and outdated information regarding the size problem. 

Quite frustrating to say the least. Is there a known method for fixing the max size limit? I don't even know, I ended up writing custom endpoints outside of trpc...

I'll leave this comment here, since this thread actually comes up in the SERPS.

**Reply by KATT** [reply]
We have no file limit built-in, if you encounter file limit issues, look at the docs of the framework you're using (nextjs/remix/fastify/etc) on how you increase the size limit on the route handler

**Reply by KATT** [reply]
It's open source, feel free to contribute with either a more specific feature request or a PR
